Shows release history when channel/arch cell is clicked in release UI

bartaz opened the pull request:
Fixes #1266 

Shows release history when channel/arch cell is clicked in release UI

### QA

- ./run or https://snapcraft-io-canonical-websites-pr-1286.run.demo.haus/
- go to Releases page of any snap
- click on any channel/arch cell
- history of releases for this channel in this architecture should show in table below

<img width="1032" alt="screen shot 2018-11-08 at 15 43 13" src="https://user-images.githubusercontent.com/83575/48205809-29618f00-e36d-11e8-93c8-accb0994c2c9.png">

webteam-app commented:
Starting demo at: https://snapcraft-io-canonical-websites-pr-1286.run.demo.haus/

therealjuan commented:
Probably we would like to show the current available release on a channel (i.e. in this case 9572 on the list):

![image](https://user-images.githubusercontent.com/8167677/48206400-42b30d00-e366-11e8-9dcf-e72a195b8fd4.png)

It would be good to have a similar hover state with a cursor pointer on unassigned and release history of a channel:
![image](https://user-images.githubusercontent.com/8167677/48206471-6b3b0700-e366-11e8-9a22-f08138b0fe3d.png)

bartaz commented:
@therealjuan I addressed the issues you mentioned.

Hover state added.

For the revisions/releases not showing in history - I removed a 100 size restriction from the API (so we get 500 past releases by default). This should cover most (if not all) current snaps to have at least currently released revisions showed properly. We should think about how to address it in the future.

bartaz commented:
'Waiting for push' button has hover state and cursor even that it cannot be clicked.

<img width="1038" alt="screen shot 2018-11-09 at 17 05 39" src="https://user-images.githubusercontent.com/83575/48273682-cbeb4200-e441-11e8-81c4-1296fbb3a392.png">

Could we make it disabled?

Lukewh commented on static/js/publisher/release/revisionsList.js:
```diff
@@ -11,35 +11,46 @@ export default class RevisionsList extends Component {
+      // disable revisions from the same architecture that already selected
+      // but only if checkboxes are visible (not in channel history)
       const isDisabled =
+        showCheckboxes &&
```
is it worth using a different variable name that's more descriptive like "inChannelHistory" that both the checkboxes and this can use?

Lukewh commented on static/js/publisher/release/revisionsList.js:
```diff
@@ -11,35 +11,46 @@ export default class RevisionsList extends Component {
-            >
-              {revision.revision}
-            </label>
+            {showCheckboxes ? (
```
again `inChannelHistory` or some such?

therealjuan commented:
Some comments:

- I was wondering if we can make the whole row of a revision clickable to toggle the checkbox from the row. Right now you can click the check box or revision number but since that is the only action you can do in the row it can be quite useful
- Maybe might be useful to showcase the proposed revision to release as part of the revision history and add it as a highlighted item. WDYT?
- Probably might need to work on some designs updated for the summary log to showcase other information more succintly (that's on me)

therealjuan commented:
Clicking the whole row makes a lot of sense. Probably we need to have another review with Lyubo once he is back to make sure that it makes sense.

For the remaining items, yes we can work them out as a separate task so you can get on with the visual updates.
